tests/foreman/ui/test_settings.py

```diff
@@ -96,125 +70,138 @@ def add_content_views_to_composite(composite_cv, org, repo):
-        result = session.settings.read('name = {}'.format(property_name))
+        session.settings.update(f'name = {property_name}', param_value)
+        result = session.settings.read(f'name = {property_name}')
         assert result['table'][0]['Value'] == param_value
 
 
 @tier2
-def test_negative_validate_error_message(session, set_original_property_value):
-    """Updates some settings with invalid values (an exceptional tier2 test)
+@pytest.mark.parametrize('setting_update', ['foreman_url', 'entries_per_page'], indirect=True)
+def test_negative_validate_foreman_url_error_message(session, setting_update):
+    """Update foreman_url and entries_per_page with invalid value (an exceptional tier2 test)
 
     :id: 7c75083d-1b4d-4744-aaa4-6fb9e93ab3c2
 
+    :parametrized: yes
+
     :expectedresults: Parameter is not updated
 
     :CaseImportance: Medium
     """
-    property_names = ['entries_per_page', 'foreman_url']
+    property_name = setting_update.name
     with session:
-        for property_name in property_names:
-            set_original_property_value(property_name)
-            for param_value in invalid_settings_values():
-                with raises(AssertionError) as context:
-                    session.settings.update('name = {}'.format(property_name), param_value)
-                assert is_valid_error_message(str(context.value))
+        invalid_value = [invalid_value for invalid_value in invalid_settings_values()][0]
+        with raises(AssertionError) as context:
+            session.settings.update(f'name = {property_name}', invalid_value)
+            assert is_valid_error_message(str(context.value))
 
 
 @run_in_one_thread
 @tier2
-def test_positive_selectors(session):
-    """"Testing input for selectors: dropdown, text area, input box
+@pytest.mark.parametrize(
+    'setting_update',
+    ['host_dmi_uuid_duplicates', 'register_hostname_fact', 'content_view_solve_dependencies'],
+    indirect=True,
+)
+def test_positive_host_dmi_uuid_duplicates(session, setting_update):
+    """"check the setting host_dmi_uuid_duplicates, register_hostname_fact &
+    content_view_solve_dependencies  value update.
 
     :id: 529ddd3a-1271-4043-9006-eac436b08b11
 
+    :parametrized: yes
+
     :expectedresults: Successfully add value to different selectors
 
     :CaseImportance: High
     """
-    uuid_input = "[ {} ]".format(gen_string("alpha"))
-    random_input = gen_string('alpha')
-    uuid_duplicate_prop = ('host_dmi_uuid_duplicates', uuid_input)
-    reg_host_prop = ('register_hostname_fact', random_input)
-    dep_solve_prop = ('content_view_solve_dependencies', 'Yes')
-
+    if setting_update.name == 'content_view_solve_dependencies':
+        property_value = "Yes"
+    else:
+        property_value = (
+            f'[ {gen_string("alpha")} ]'
+            if setting_update.name == 'host_dmi_uuid_duplicates'
+            else gen_string('alpha')
+        )
```

Comment:
Personally, I find previous version much more readable. We have three settings and each require different value:
* `content_view_solve_dependencies` takes `"Yes"`
* `register_hostname_fact` takes `gen_string('alpha')`
* `host_dmi_uuid_duplicates` takes `f'[ {gen_string("alpha")} ]'`

Couldn't we just make a map of setting names and their values, instead of multi-level if statement?

Reply:
I think this is much better.
```
 property_dict = {"content_view_solve_dependencies": "Yes", "host_dmi_uuid_duplicates":
                     f'[ {gen_string("alpha")} ]', "register_hostname_fact": gen_string('alpha')}
```